core/modules/flowgen.cc

```diff
@@ -231,6 +235,72 @@ pb_error_t FlowGen::InitFlowPool() {
   return pb_errno(0);
 }
 
+void FlowGen::UpdateDerivedParameters() {
+  /* calculate derived variables */
+  pareto_.inversed_alpha = 1.0 / pareto_.alpha;
+
+  if (duration_ == DURATION_PARETO) {
+    MeasureParetoMean();
+  }
+
+  concurrent_flows_ = flow_rate_ * flow_duration_;
+  if (concurrent_flows_ > 0.0) {
+    flow_pps_ = total_pps_ / concurrent_flows_;
+  }
+
+  flow_pkts_ = flow_pps_ * flow_duration_;
+  if (flow_rate_ > 0.0) {
+    flow_gap_ns_ = 1e9 / flow_rate_;
+  }
+}
+
+pb_cmd_response_t FlowGen::CommandUpdate(const bess::pb::FlowGenArg &arg) {
```

Comment:
Aren't the various calls to process arg mostly the same as in Init()?  If so, could more of this be folded into UpdateDerivedParameters()?